src/WebJobs.Script/Workers/Rpc/Configuration/WorkerProfileConditionFactory.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers
+{
+    internal class WorkerProfileConditionFactory : IWorkerProfileConditionFactory
+    {
+        private readonly ILogger _logger;
+        private readonly ISystemRuntimeInformation _systemRuntimeInformation;
+        private readonly IEnvironment _environment;
+
+        public WorkerProfileConditionFactory(ILogger logger, ISystemRuntimeInformation systemRuntimeInfo, IEnvironment environment)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _systemRuntimeInformation = systemRuntimeInfo ?? throw new ArgumentNullException(nameof(systemRuntimeInfo));
+        }
+
+        public IWorkerProfileCondition CreateWorkerProfileCondition(string type, string name, string expression)
+        {
+            Enum.TryParse(type, out ConditionType conditionType);
+            switch (conditionType)
+            {
+                case ConditionType.HostProperty:
```

Comment:
This logic should be directly on the string type. The exception below should only be thrown if no condition provider matches the type